src/components/Button/Button.tsx

```diff
@@ -157,23 +157,36 @@ function Button({
 
   const type = submit ? 'submit' : 'button';
 
-  return url ? (
-    <UnstyledLink
-      id={id}
-      url={url}
-      external={external}
-      onClick={onClick}
-      onFocus={onFocus}
-      onBlur={onBlur}
-      onMouseUp={handleMouseUpByBlurring}
-      className={className}
-      disabled={isDisabled}
-      aria-label={accessibilityLabel}
-    >
-      {indicatorMarkup}
-      {content}
-    </UnstyledLink>
-  ) : (
+  if (url) {
+    return isDisabled ? (
+      // <a disabled> is not valid HTML so render an <a> without a href attribute.
+      // This results in a minimal DOM change when the Button moves between
+      // enabled and disabled states.
+      // eslint-disable-next-line jsx-a11y/anchor-is-valid
+      <a id={id} className={className} aria-label={accessibilityLabel}>
```

Comment:
Would this still appear as a navigable anchor in the accessibility tree (just like a good old `<a name="foo"></a>`)?